R/utils.R

```diff
@@ -18,20 +20,25 @@ get_vrd_key <- function() {
 #'
 #'helper function for querying api database
 #'
+#' @param url_ api url to be hit
+#' @param query names the resulting object
+#' @param limit how many items to get from the api, default to no limit
+#' @param api_key defaults to the one set in the environment, can be pass one manually
+#'
 #' @return vrd_api class
 #' @export
 #'
 #' @examples
 #' \dontrun{
 #' call_vrd_api()
 #' }
-call_vrd_api <- function(url_, query, limit = NULL){
+call_vrd_api <- function(url_, query, limit = NULL, api_key=get_vrd_key()){
```

Comment:
defaults to calling the get api key function

sane default behaviour, and makes it easy to test